crates/host/src/wasmbus/mod.rs

```diff
@@ -318,79 +336,39 @@ struct Handler {
     origin: WasmCloudEntity,
     // package -> target -> entity
     links: Arc<RwLock<HashMap<String, HashMap<String, WasmCloudEntity>>>>,
-    /// The current link name to use for interface targets, overridable in actor code via set_target()
+    /// The current link name to use for interface targets, overridable in actor code via set_link_name()
     interface_link_name: Arc<RwLock<String>>,
-    // link name -> package -> interface -> target
+    // link name -> package -> interface -> component ID
     #[allow(clippy::type_complexity)]
-    interface_links: Arc<RwLock<HashMap<String, HashMap<String, HashMap<String, String>>>>>,
-    // NOTE(brooksmtownsend): needs deduplication of responsibility with interface_links
-    wrpc_targets: Arc<RwLock<HashMap<TargetInterface, TargetEntity>>>,
-    aliases: Arc<RwLock<HashMap<String, WasmCloudEntity>>>,
+    interface_links: Arc<
+        RwLock<HashMap<LinkName, HashMap<WitNsAndPackage, HashMap<WitInterface, ComponentId>>>>,
+    >,
     chunk_endpoint: ChunkEndpoint,
 }
 
-#[instrument(level = "trace")]
-async fn resolve_target(
-    target: Option<&TargetEntity>,
-    links: Option<&HashMap<String, WasmCloudEntity>>,
-    aliases: &HashMap<String, WasmCloudEntity>,
-) -> anyhow::Result<WasmCloudEntity> {
-    const DEFAULT_LINK_NAME: &str = "default";
-
-    trace!("resolve target");
-
-    let target = match target {
-        None => links
-            .and_then(|targets| targets.get(DEFAULT_LINK_NAME))
-            .context("link not found")?
-            .clone(),
-        Some(TargetEntity::Wrpc(target)) => {
-            let (namespace, package, _) = target.interface.as_parts();
-            WasmCloudEntity {
-                public_key: target.id.clone(),
-                contract_id: format!("{namespace}:{package}"),
-                link_name: target.link_name.clone(),
-            }
-        }
-        Some(TargetEntity::Link(link_name)) => links
-            .and_then(|targets| targets.get(link_name.as_deref().unwrap_or(DEFAULT_LINK_NAME)))
-            .context("link not found")?
-            .clone(),
-        Some(TargetEntity::Actor(ActorIdentifier::Key(key))) => WasmCloudEntity {
-            public_key: key.public_key(),
-            ..Default::default()
-        },
-        Some(TargetEntity::Actor(ActorIdentifier::Alias(alias))) => aliases
-            .get(alias)
-            .context("unknown actor call alias")?
-            .clone(),
-    };
-    Ok(target)
-}
-
 impl Handler {
     #[instrument(level = "debug", skip(self, operation, request))]
     async fn call_operation_with_payload(
         &self,
-        target: Option<TargetEntity>,
+        target: Option<WrpcTarget>,
         operation: impl Into<String>,
         request: Vec<u8>,
     ) -> anyhow::Result<Result<Vec<u8>, String>> {
-        let links = self.links.read().await;
-        let aliases = self.aliases.read().await;
+        // todo(vadossi-cosmonic): handle aliases...?
```

Comment:
Good news, we don't need em anymore! Call aliases were useful because we didn't have actor links, but now we do. Instead of looking up the call alias, we'll look up the target actor based on link name the same way we do providers.

Reply:
Great! Will remove this TODO

- [ ] Remove TODO around actor aliases